core/p2p/p2pv1/conn.go

```
@@ -0,0 +1,103 @@
+package p2pv1
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"sync"
+
+	"github.com/pkg/errors"
+	log "github.com/xuperchain/log15"
+	p2pPb "github.com/xuperchain/xuperchain/core/p2p/pb"
+	p2p_pb "github.com/xuperchain/xuperchain/core/p2p/pb"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// Conn maintain the connection of a node
+type Conn struct {
+	// addr:"IP:Port"
+	id     string
+	lg     log.Logger
+	cli    p2p_pb.P2PService_SendP2PMessageClient
+	quitCh chan bool
+	lock   *sync.RWMutex
+}
+
+func NewConn(lg log.Logger, addr string, certPath, serviceName string, maxMsgSize int) *Conn {
+	conn := &Conn{
+		id:     addr,
+		lg:     lg,
+		quitCh: make(chan bool, 1),
+		lock:   &sync.RWMutex{},
+	}
+	conn.NewGrpcClient(maxMsgSize, certPath, serviceName)
+	return conn
+}
+
+func genCreds(certPath, serviceName string) (credentials.TransportCredentials, error) {
+	bs, err := ioutil.ReadFile(certPath + "/cert.crt")
+	if err != nil {
+		return nil, err
+	}
+	certPool := x509.NewCertPool()
+	ok := certPool.AppendCertsFromPEM(bs)
+	if !ok {
+		return nil, err
+	}
+
+	certificate, err := tls.LoadX509KeyPair(certPath+"/key.pem", certPath+"/private.key")
```

can ignore err here ?

---

thanks, fixed